src/components/Dropdown/Dropdown.tsx

```diff
@@ -30,26 +36,69 @@ const Dropdown: React.FC<Props> = ({
   onChange = () => null,
   defaultOption,
   className,
+  tag,
+  onClick,
   ...rest
 }) => {
   const onDropdownChange = (e: ChangeEvent<HTMLSelectElement>) => {
     if (typeof onChange === 'function') {
       onChange(e.target.value);
     }
   };
+
   return (
-    <DropdownContainer onChange={onDropdownChange} className={className} {...rest}>
-      {defaultOption && (
-        <option value="" disabled={defaultOption.disabled}>
-          {defaultOption.displayName}
-        </option>
+    <AnalyticsComponent>
+      {({ pages, tracker, merchantId, ip }) => (
+        <DropdownContainer
+          onChange={
+            !tag
+              ? onDropdownChange
+              : (e) => {
+                  if (onDropdownChange) onDropdownChange(e);
+
+                  const selectedValue = e.currentTarget.selectedIndex;
+                  tracker.track({
+                    event: AnalyticsEvents.DROPDOWN_CHANGE,
+                    type: AnalyticsComponentType.DROPDOWN,
+                    tag: `dropdown_${snakeCase(e.currentTarget.options[selectedValue].text)}`,
+                    pages,
+                    merchantId,
+                    ip,
+                  });
+                }
+          }
```

Comment:
Normally I would be against inline function props, but you're already inside a function with its own arguments and extracting this function into a separate const would mean defining it in a scope where `pages`, `tracker`, `merchantId` and `ip` don't exist, so it's fine leaving it like this.

But how about making the following little change and incorporate the `!tag` condition inside the event handler:

```suggestion
          onChange={(e) => {
            onDropdownChange?.(e);
    
            if (tag) {
              const selectedValue = e.currentTarget.selectedIndex;
              tracker.track({
                event: AnalyticsEvents.DROPDOWN_CHANGE,
                type: AnalyticsComponentType.DROPDOWN,
                tag: `dropdown_${snakeCase(e.currentTarget.options[selectedValue].text)}`,
                pages,
                merchantId,
                ip,
              });
            }
          }}
```